CP-CPS.md

```diff
@@ -1089,14 +1072,23 @@ See section 7.1.
 
 #### 7.1.3.1 SubjectPublicKeyInfo
 
-No stipulation.
+The `AlgorithmIdentifier` field of the `SubjectPublicKeyInfo` field of ISRG Certificates has the following (hex-encoded) DER bytes, corresponding to the key type:
+
+| Type        | Hex AlgorithmIdentifier                      |
+| ----------- | -------------------------------------------- |
+| RSA         | `300d06092a864886f70d0101010500`             |
+| ECDSA P-256 | `301306072a8648ce3d020106082a8648ce3d030107` |
+| ECDSA P-384 | `301006072a8648ce3d020106052b81040022`       |
+| ECDSA P-521 | `301006072a8648ce3d020106052b81040023`       |
 
 #### 7.1.3.2 Signature AlgorithmIdentifier
 
-| Name                    | Object identifier                    |
-| ----------------------- | ------------------------------------ |
-| sha256WithRSAEncryption | 1.2.840.113549.1.1.11                |
-| ecdsa-with-SHA384       | 1.2.840.10045.4.3.3                  |
+The `AlgorithmIdentifier` field of `signature` and `signatureAlgorithm` fields of ISRG Certificates, CRLs, and OCSP Responses has the following (hex-encoded) DER bytes, corresponding to the issuer's key type and signature algorithm:
```

Comment:
BRs 7.1.3.2:

> 7.1.3.2 Signature AlgorithmIdentifier
> All objects signed by a CA Private Key MUST conform to these requirements on the use of the
> AlgorithmIdentifier or AlgorithmIdentifier‐derived type in the context of signatures.
> In particular, it applies to all of the following objects and fields:
> • The signatureAlgorithm field of a Certificate or Precertificate.
> • The signature field of a TBSCertificate (for example, as used by either a Certificate or
Precertificate).
> • The signatureAlgorithm field of a CertificateList
> • The signature field of a TBSCertList
> • The signatureAlgorithm field of a BasicOCSPResponse.

The language in this draft misses some objects / fields. We should mirror the BRs:

```suggestion
The encoding of the `AlgorithmIdentifier` field of `signature` and `signatureAlgorithm` fields of the following objects and fields uses the hex-encoded DER bytes specified below, corresponding to key type:

 • The signatureAlgorithm field of a Certificate or Precertificate.
 • The signature field of a TBSCertificate (for example, as used by either a Certificate or
Precertificate).
 • The signatureAlgorithm field of a CertificateList
 • The signature field of a TBSCertList
 • The signatureAlgorithm field of a BasicOCSPResponse.
```

Also the list of key type / algorithm identifier should include ECDSA P-256. We don't currently have any P-256 intermediates but we can specify which AlgorithmIdentifier we would use if we did have them.

Reply:
I disagree that anything has been missed: "`signature` and `signatureAlgorithm` fields of ISRG Certificates" captures the first two bullet points as both of those fields are in Certificates, and Precertificates are a kind of Certificate. The same holds for the two bullet points representing CRLs and the one bullet point covering OCSP.

That said, listing these doesn't actually help: the five bullet points in BRs 7.1.3.2 are actually examples, and the requirement is on "All objects signed by a CA" so maybe we should just say that?

I don't think we should specify the AlgorithmIdentifier we would use for P-256 intermediates, unless we also update Section 6.1.5 to say that we have P-256 intermediates.

Reply:
Yeah, I like just saying "all objects signed by the CA."